src/errors/location.rs

```diff
@@ -205,9 +205,11 @@ impl Serialize for Location {
 impl TryFrom<Option<&PyAny>> for Location {
     type Error = PyErr;
 
+    /// Only ever called by ValidationError -> PyLineError to convert user input to our internal Location
+    /// Thus this expects the location to *not* be reversed and reverses it before storing it.
     fn try_from(location: Option<&PyAny>) -> PyResult<Self> {
```

Comment:
I think if we have things like this that are only ever called from one place it might be best to just implement them in that one place instead of a reusable trait to avoid confusion around this sort of implementation details stemming from how the trait ends up getting used.